src/App.tsx

```diff
@@ -16,35 +22,96 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const handleReset = () => {
+    setSortField(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
+  const getSortedGoods = () => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortField === SortType.ALPHABETICALLY) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortField === SortType.BY_LENGTH) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
```

Comment:
[PERFORMANCE] - The `getSortedGoods` function is called on every render, which could lead to performance issues if the list of goods is large. Consider using `useMemo` to only compute the sorted goods when `sortField` or `isReversed` changes.